WordPress/gutenberg
# Font Library: Improve font collection rest controller

## What?
- Improves font collection rest controller to look and work more similarly to other core components.
- Makes pagination work to get font collections
- Makes _fields work 
- Optimizes the external data request. I loads json files only when `font_families` or `categories` are requested.
- Updates the fields requested from the frontend to the API.

## Why?
- To make it work more similarly to other core components.
- To API users/extenders more control about the data requested.

## How?
Changing the REST controller.

## Testing Instructions
- Run unit tests
- Check that the font collection functionality still works as expected.
